### PR TITLE
Feat: helm repo as addon registry to support addon's  multi-version

### PR DESCRIPTION
**Pull request by wangyikewxgm**
helm repo as addon registry to support addon's  multi-version

Signed-off-by: 楚岳 <wangyike.wyk@alibaba-inc.com>

### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/addon/cache.go [hunk 1]
```diff
@@ -210,17 +257,36 @@ func (u *Cache) discoverAndRefreshRegistry() {
...
+			}
+			for _, addon := range uiDatas {
+				uiData, err := versionedRegistry.GetAddonUIData(context.Background(), addon.Name, addon.Version)
+				if err != nil {
+					log.Logger.Errorf("fail to get addon from registry %s, addon %s version %s for cache updating, %v", addon.Name, r.Name, err)
+					continue
+				}
+				u.putVersionedUIData2Cache(r.Name, addon.Name, addon.Version, uiData)
+				// we also no version key, if use get addonUIData without version will return this vale as latest data.
+				u.putVersionedUIData2Cache(r.Name, addon.Name, "latest", uiData)
```
Comment (wonderflow) [inline pkg/addon/cache.go:288]: how do you know which one is the latest when there're many versions?

Comment (wangyikewxgm) [reply pkg/addon/cache.go:288]: listAddon will use `sort.Sort(sort.Reverse(versions))` to make latest addon as the first one.

**Review by wonderflow — COMMENTED** [review]
please increase the test coverage to match the requirement

**Review by wonderflow — APPROVED** [review]
as talked with @wangyikewxgm offline, he has promised to add unit-tests and refactor the code in the following PRs, let's merge it first as it blocks some other features.

**Review by barnettZQG — APPROVED** [review]

**Pull request merged**